server/src/main/java/io/crate/execution/engine/aggregation/impl/TopKAggregation.java

```diff
@@ -436,13 +470,41 @@ int id() {
         }
     }
 
+    private static boolean supportedByLongSketch(DataType<?> type) {
+        return switch (type.id()) {
+            case LongType.ID -> true;
+            case DoubleType.ID -> true;
+            case FloatType.ID -> true;
+            default -> false;
+        };
+    }
+
+    private static long toLong(DataType<?> type, Object o) {
+        return switch (type.id()) {
+            case LongType.ID -> (Long) o;
+            case DoubleType.ID -> NumericUtils.doubleToSortableLong((Double) o);
+            case FloatType.ID -> (long) NumericUtils.floatToSortableInt((Float) o);
+            default -> throw new IllegalArgumentException("Type cannot be converted to long");
+        };
+    }
+
+    private static Object toObject(DataType<?> type, long o) {
```

Comment:
can we use `Number` instead of `Object` here?

Reply:
It's type Object because there can be types which can be processed as long which are not Number like Timestamp without Timezone.